### DISCUSSION TITLE
COMPATIBILITY_CONTEXT_ELEMENT breaking change in 10.0.18362.0

### DISCUSSION THREAD

**Discussion by KalleOlaviNiemitalo** [microsoft/win32metadata#661] [Q&A]
The COMPATIBILITY_CONTEXT_ELEMENT structure in \<winnt.h> got a new member `ULONGLONG MaxVersionTested` between Windows SDK versions 10.0.17763.0 and 10.0.18362.0. This changed sizeof(COMPATIBILITY_CONTEXT_ELEMENT) from 20 to 32. The ACTIVATION_CONTEXT_COMPATIBILITY_INFORMATION structure has a trailing array `COMPATIBILITY_CONTEXT_ELEMENT Elements[]`, which cannot be correctly indexed if the element size does not match. \<winnt.h> does not make the earlier versions of these structures available in any way.

Changes like that should hopefully be rare. If they do occur, should win32metadata publish both definitions, even though the headers don't? 

[More details about the change on an archived MSDN forum](<https://social.msdn.microsoft.com/Forums/en-US/8d128a1e-a0b8-46b9-b1fb-a8ceb7b7442a/compatibilitycontextelement-breaking-change-in-100183620> "\"COMPATIBILITY_CONTEXT_ELEMENT breaking change in 10.0.18362.0\", posted to the \"Application Compatibility for Windows Desktop Development\" MSDN forum on 2019-08-27").

**Comment by mikebattista** [comment]
@sotteson1 what are your thoughts on this?
